### PR TITLE
feat: PC-12014 Support CompositesV2

### PR DESCRIPTION
**Pull request by BSski**

### CODE REVIEW

**Comment by nikodemrafalski** [comment]
The documentation needs to be regenerated with `make generate` and committed. Also, just double-checking, have the documentation  changes been reviewed by the tech writing team?

**Comment by annatrifonova** [comment]
Findings after the 1st round of tests:
1. Objective's `display_name` is required now - should be optional (this is not related to the changes made for composites, but after discussion with Bartek decided to fix it here)
2. `composite` block cannot be applied without `objectives` and `components` - they should be optional (same as in sloctl)
3. `objectives` block cannot be empty (now terraform provider is craching when it's empty) - `objectives` should be optional

File: nobl9/resource_slo.go [hunk 1]
```diff
@@ -116,6 +116,12 @@ func resourceObjective() *schema.Resource {
 					},
 				},
 			},
+			"composite_v2": {
```
[the remark this answers is not in the document]
Comment (BSski) [reply nobl9/resource_slo.go]: I removed the suffix after our discussion. Thanks!

File: nobl9/resource_slo.go [hunk 2]
```diff
@@ -116,6 +116,12 @@ func resourceObjective() *schema.Resource {
...
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "A composite SLO aggregates SLOs to measure their overall performance.",
```
[the remark this answers is not in the document]
Comment (BSski) [reply nobl9/resource_slo.go]: Good point, thanks. I consulted it with tech writers and introduced the changes.

File: docs/resources/slo.md [hunk 3]
```diff
@@ -145,67 +145,95 @@ resource "nobl9_slo" "this" {
...
+<a id="nestedblock--objective--composite--components--objectives--composite_objective"></a>
+### Nested Schema for `objective.composite.components.objectives.composite_objective`
+
+Required:
+
+- `objective` (String) SLO objective name.
+- `project` (String) Project name.
+- `slo` (String) SLO name.
+- `weight` (Number) Weights determine each component’s contribution to the composite SLO.
+- `when_delayed` (String) Indicator of how to interpret data received after max_delay is over.
```
Comment (nikodemrafalski) [inline docs/resources/slo.md:224]: The data might never been received even after the delay - so the story is not about delayed data it's about missing data

```suggestion
- `when_delayed` (String) An indicator of how to interpret the component not receiving any data before `max_delay' elapses
```

File: nobl9/resource_slo.go [hunk 4]
```diff
@@ -157,6 +163,79 @@ func resourceObjective() *schema.Resource {
...
+			},
+			"weight": {
+				Type:        schema.TypeFloat,
+				Required:    true,
+				Description: "Weights determine each component’s contribution to the composite SLO.",
+			},
+			"when_delayed": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Indicator of how to interpret data received after max_delay is over.",
```
Comment (nikodemrafalski) [inline nobl9/resource_slo.go:232]: Same comment as above in docs section

File: docs/resources/slo.md (hunk 3 above)
Comment (BSski) [reply docs/resources/slo.md]: Thanks, I missed that. Consulted it with tech writers again and now it's `Defines how to treat missing component data on 'max_delay' expiry.`. Is it good in your opinion?

**Review by nikodemrafalski — APPROVED** [review]

**Pull request merged**